server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/configuration/CassandraMailQueueViewConfiguration.java

```diff
@@ -67,6 +69,30 @@ public CassandraMailQueueViewConfiguration build() {
         return bucketCount -> updateBrowseStartPace -> sliceWindow -> new Builder.ReadyToBuild(bucketCount, updateBrowseStartPace, sliceWindow);
     }
 
+    public static CassandraMailQueueViewConfiguration from(Configuration configuration) {
+        Integer bucketCount = configuration.getInteger(BUCKET_COUNT_PROPERTY, null);
+        Preconditions.checkState(bucketCount != null, "You need to specify the " +
+            BUCKET_COUNT_PROPERTY + " property as the number of nodes from your cassandra cluster");
+
+        Integer updateBrowseStartPace = configuration.getInteger(UPDATE_BROWSE_START_PACE_PROPERTY, null);
+        Preconditions.checkState(updateBrowseStartPace != null, "You need to specify the " +
+            UPDATE_BROWSE_START_PACE_PROPERTY + " property as the average number of en-queue times, then after that, updating cassandra browse start point");
+
+        Long sliceWindowInSecond = configuration.getLong(SLICE_WINDOW_PROPERTY, null);
```

Comment:
No this is not fiendly enough.

IMO we need to handle units ( and not a second count). Please handle this as a String and have a look to **TimeConverter** class